gateway-provider-security-jwt/src/main/java/org/apache/knox/gateway/provider/federation/jwt/filter/AbstractJWTFilter.java

```diff
@@ -187,6 +186,22 @@ protected boolean tokenIsStillValid(JWT jwtToken) throws UnknownTokenException {
     return expires == null || new Date().before(expires);
   }
 
+  protected boolean tokenIsStillValid(final String tokenId) throws UnknownTokenException {
+    Date expires = getExpiration(tokenId);
+    return expires == null || (new Date().before(expires));
+  }
+
+  private Date getExpiration(final String tokenId) throws UnknownTokenException {
```

Comment:
We may indicate - in the method name - that expiration is checked in token state service.

Reply:
Changed to getServerManagedStateExpiration(String)